### PR TITLE
refactor(typescript): Convert TableHead, TableHeader to tsx

### LINKED ISSUE #12527: Add TypeScript types to TableHead

### PR DESCRIPTION
**Pull request by pratikkarad**
Closes #12527, #12528

Converts TableHead and TableHeader to tsx in preparation for the main Table conversion

#### Changelog

**Changed**

- Converted TableHead.js -> TableHead.tsx
- Converted TableHeader.js -> TableHeader.tsx

#### Testing / Reviewing

To verify these changes are working, we can check to see if the existing storybook behavior is still working correctly. Should be able to properly import converted components in a TS environment

### CODE REVIEW

File: packages/react/src/components/DataTable/TableHead.tsx [hunk 1]
```diff
@@ -5,9 +5,12 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import React, { ThHTMLAttributes } from 'react';
 import wrapComponent from '../../tools/wrapComponent';
 
-const TableHead = wrapComponent({
+export type TableHeadProps = ThHTMLAttributes<HTMLTableCellElement>;
```
Comment (francinelucca) [inline packages/react/src/components/DataTable/TableHead.tsx]: should this be an `HTMLTableSectionElement` instead? https://developer.mozilla.org/en-US/docs/Web/API/HTMLTableSectionElement. Doesn't look like `thead` is considered an HTMLTableCellElement https://developer.mozilla.org/en-US/docs/Web/API/HTMLTableCellElement
```suggestion
export type TableHeadProps = ThHTMLAttributes<HTMLTableSectionElement>;
```

**Review by francinelucca — CHANGES_REQUESTED** [review]
Thanks for putting this together @pratikkarad! Just one change request regarding the `thead` interface. also looks like some of the CI checks are failing

File: packages/react/src/components/DataTable/TableHead.tsx (hunk 1 above)
Comment (pratikkarad) [reply packages/react/src/components/DataTable/TableHead.tsx]: Thanks for pointing it out. I have updated the code.

File: packages/react/src/components/DataTable/TableHeader.tsx [hunk 2]
```diff
@@ -91,7 +158,7 @@ const TableHeader = React.forwardRef(function TableHeader(
     [`${prefix}--table-sort--descending`]:
       isSortHeader && sortDirection === sortStates.DESC,
   });
-  const ariaSort = !isSortHeader ? 'none' : sortDirections[sortDirection];
+  const ariaSort = !isSortHeader ? 'none' : sortDirection ? sortDirections[sortDirection] : sortDirections[sortStates.NONE];
```
Comment (pratikkarad) [inline packages/react/src/components/DataTable/TableHeader.tsx]: I observed It is still failing here. I tried debugging but got no luck. Any help or suggestion is appreciated. 😃

Comment (francinelucca) [reply packages/react/src/components/DataTable/TableHeader.tsx]: let's change this line to ` const ariaSort = (!isSortHeader || !sortDirection) ? 'none' : sortDirections[sortDirection];`

and change the type of `sortDirections` in the top to 
```
const sortDirections: {[key: string]: 'none' | 'ascending' | 'descending'} = {
  [sortStates.NONE]: 'none',
  [sortStates.ASC]: 'ascending',
  [sortStates.DESC]: 'descending',
};
```


That should fix this issue

**Review by pratikkarad — COMMENTED** [review]
Hi @francinelucca, I'm facing a problem with declaring the static properties on the functional component in typescript.
For eg.
`TableHeader.translationKeys = Object.values(translationKeys);`
In TableHeader, `translationKeys` is giving the compilation error.
I tried creating a separate type as suggested on StackOverflow/blogs, but still got no luck.

**Comment by francinelucca** [comment]
> Hi @francinelucca, I'm facing a problem with declaring the static properties on the functional component in typescript. For eg. `TableHeader.translationKeys = Object.values(translationKeys);` In TableHeader, `translationKeys` is giving the compilation error. I tried creating a separate type as suggested on StackOverflow/blogs, but still got no luck.

can you try doing `(TableHeader as any).translationKeys = Object.values(translationKeys);`

**Review by francinelucca — APPROVED** [review]

**Review by aledavila — APPROVED** [review]

**Pull request merged**